terraform/modules/aws-network-load-balancer/README.md

```diff
@@ -0,0 +1,37 @@
+Terraform module: Network load balancer
+=======================================
+
+State: __experimental__
+
+This module creates a network load balancer for HTTP (port 80) and HTTPS (port 443) traffic.
+It uses a *target group* for each port and attaches all instances that share the given *role*
+to each group. It furthermore uses the given target ports to check their health.
+
+Load balancing happens across availability zones. The VPC is determined by the given environment.
+The subnets used within the VPC are assumed to
+
+a) have an internet gateway
+b) are tagged with `Routability:public`
+c) be attached to the machines tagged with the `target_role`
+
+
+*Please note, in order for this to work, ingress has to be allowed on the given target ports for all target machines.
+Furthermore, since those target machines are referred to by their role (AWS tag) and are not part of an auto-scaling
+group, the instance of this module has to be re-applied every time the set of machines changes.* 
+
+AWS resources: lb (type: network)
+
+
+#### How to use the module
+
+```hcl
+module "nlb" {
+  source = "github.com/wireapp/wire-server-deploy.git//terraform/modules/aws-network-load-balancer?ref=develop"
```

Comment:
While only an example, this is likely getting copy-pasted into different locations (if we don't do something like https://github.com/zinfra/backend-issues/issues/1576 or the other linked issues)

We should never refer to a mutable ref, but a specific commit ID for reproducibility.

Reply:
I'd just change this to have `?ref=$commitID`, and defer the lookup and updating strategy to the ones using the module.

Reply:
As you said yourself, it's an example and likely to be copied, so, changing this to `$commitID` would become a non-working example. While I'm pro putting people's nose into this when copy&pasting w/o thinking, I would propose

1) do this in a different PR, because it's done this way in all the other modules, too
2) choose sth less misleading, e.g. `?ref=CHANGE-ME` because a ref can also be sth else, e.g. tag or branch, not just a commit ID. People can make their own choice when it comes to version pinning.

Would you be okay with that?

Reply:
Yeah, my whole point was to simply not silently/accidentially pin this to a branch.

I don't really have a strong opinion on whether this should be `CHANGE-ME` or `$commitID`, pick what you prefer. :+1:

Reply:
This still needs to be replaced with some placeholder.

Reply:
The cleanup PR removing the other occurences of where we just propose using `develop` is here: https://github.com/wireapp/wire-server-deploy/pull/318

Reply:
The PR was merged, please update the placeholder to be consistent with the other locations.

Reply:
Doubling the effort. sure why not. Adjusted.